storage/rocksdb/src/block.rs

```diff
@@ -114,24 +124,19 @@ impl BlockQueries for RocksDbBlockQueries {
         }))
     }
 
-    fn by_height(
-        &self,
-        db: &Self::Storage,
-        height: u64,
-        include_transactions: bool,
-    ) -> Result<Option<BlockResponse>, Self::Err> {
-        db.get_pinned_cf(&height_cf(db), height.to_key())?
-            .map(|hash| B256::from_slice(hash.as_ref()))
-            .map(|hash| self.by_hash(db, hash, include_transactions))
-            .unwrap_or(Ok(None))
+    fn get_forkchoice_state(&self, db: &Self::Storage) -> Result<ForkchoiceState, Self::Err> {
+        let state = db
+            .get_pinned_cf(&fc_cf(db), FC_KEY)?
+            .map(|v| ForkchoiceState::from_value(&v))
+            .unwrap_or_default();
+        Ok(state)
     }
 
-    fn latest(&self, db: &Self::Storage) -> Result<Option<u64>, Self::Err> {
-        Ok(db
-            .iterator_cf(&height_cf(db), IteratorMode::End)
-            .next()
-            .transpose()?
-            .map(|(height, _)| u64::from_key(height.as_ref())))
+    fn height_to_hash(&self, db: &Self::Storage, height: u64) -> Result<B256, Self::Err> {
+        let maybe_hash = db
+            .get_pinned_cf(&fc_cf(db), height.to_key())?
```

Comment:
Shouldn't we be querying height CF here? And AFAICT forkchoice is keyed by a single string const anyways.

Reply:
Fixed in https://github.com/UmiNetwork/op-move/pull/563/commits/9b73e68560267e8f0bdbf4ba256b86b4739d7124